common/src/main/java/io/netty/util/internal/MpscLinkedQueue.java

```diff
@@ -248,53 +256,37 @@ public E element() {
         throw new NoSuchElementException();
     }
 
-    @Override
-    public Object[] toArray() {
-        final Object[] array = new Object[size()];
-        final Iterator<E> it = iterator();
-        for (int i = 0; i < array.length; i ++) {
-            if (it.hasNext()) {
-                array[i] = it.next();
-            } else {
-                return Arrays.copyOf(array, i);
+    private List<E> toList(int initialCapacity) {
+        List<E> elements = new ArrayList<E>(initialCapacity);
+
+        MpscLinkedQueueNode<E> n = peekNode();
+        for (;;) {
+            if (n == null) {
+                break;
             }
+            E value = n.value();
+            if (value == null) {
+                break;
+            }
+            elements.add(value);
+            MpscLinkedQueueNode<E> next = n.next();
+            if (n == next) {
+                break;
+            }
+            n = next;
         }
-        return array;
+        return elements;
+    }
+
+    @Override
+    public Object[] toArray() {
+        return toList(16).toArray();
```

Comment:
@Scottmitch The 16 came from the mighty @normanmaurer =P Joking aside I just thought it may not be a bad fit as queues often have some items in there. That said I not care too much about this method as we not use any of these methods that call it in our code-base.